provider/resource_keycloak_group_memberships_test.go

```diff
@@ -34,6 +34,23 @@ func TestAccKeycloakGroupMemberships_basic(t *testing.T) {
 	})
 }
 
+func TestAccKeycloakGroupMemberships_moreThan100members(t *testing.T) {
+	realmName := "terraform-" + acctest.RandString(10)
+	groupName := "terraform-group-" + acctest.RandString(10)
+	username := "terraform-user-101"
+
+	resource.Test(t, resource.TestCase{
+		Providers: testAccProviders,
+		PreCheck:  func() { testAccPreCheck(t) },
+		Steps: []resource.TestStep{
+			{
+				Config: testKeycloakGroupMemberships_moreThan100members(realmName, groupName),
+				Check:  testAccCheckUserBelongsToGroup("keycloak_group_memberships.group_members", username),
```

Comment:
The `Check` doesn't hurt but it actually isn't needed for this test.  Terraform tests will automatically perform a `plan` after applying the config, and if the plan shows a change, it will automatically fail the test.

Before your changes, this test (without the `Check`) would fail because the old code would only fetch 100 members from Keycloak, and the Terraform state thinks there are >100 members after applying, so Terraform will think it needs to add the missing members.

The changes you introduced will correctly fetch all of the group's members, so the subsequent plan after applying will show no changes.

Anyways, I'd feel better about removing this since it seems strange to only check a single member.

Reply:
Removed check clause